PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -424,6 +424,32 @@ def miniAOD_customizeCommon(process):
     delattr(process, 'selectedUpdatedPatJetsPuppiJetSpecific')
 
     task.add(process.slimmedJetsPuppi)
+
+    # Embed pixelClusterTagInfos in slimmedJets
+    process.patJets.addTagInfos = cms.bool(True)
+    process.patJets.tagInfoSources.append( cms.InputTag("pixelClusterTagInfos") )
```

Comment:
shouldn't this be done with the standard btag tools? PhysicsTools/PatAlgos/python/tools/jetTools.py

Reply:
A line in jetTools.py to include pixelClusterTagInfos has been added since the beginning (L595 in jetTools.py)
In principle, it's not necessary to use the jetTool or perform additional updateJetCollection steps: everything we need is already there, we just need to set the existing modules to keep the pixelClusterTagInfos and not discard them.

Reply:
This is somewhat in collision with changes made in ` PhysicsTools/PatAlgos/python/slimming/MicroEventContent_cff.py`. How do you plan to access these TagInfos later on? If you plan to access them directly, then adding `'keep *_pixelClusterTagInfos_*_*'` is fine and you don't need to set `dropTagInfos = '0'` and `addTagInfos = True`. In fact, all of this addition of TagInfos is for nothing because they are later dropped by the output module because there is no explicit keep statement for them. However, if they were kept, there would be a big jump in the event size because you are not only adding pixelClusterTagInfo (note the `tagInfoSources.append( cms.InputTag("pixelClusterTagInfos") )` call) but all other TagInfos.

If you really need to add TagInfo to slimmed jets, keep in mind that they are really stored in a separate collection (the name starts with `recoBaseTagInfosOwned_...`) so in that case you would have to ensure that that particular collection of TagInfos is kept, that you only keep pixel cluster TagInfos (so no appending) and you would also need to implement an appropriate TagInfo accessor method in `pat::Jet` similar to those in https://github.com/cms-sw/cmssw/blob/CMSSW_10_6_2/DataFormats/PatCandidates/interface/Jet.h#L198-L210.

Reply:
Thanks Dinko for the suggestion. Indeed, the plan is to use the second method (be able to  get the Tag Infos from slimmedJets). In the latest commit, we drop `*_pixelClusterTagInfos_*_*`, but we now keep `recoBaseTagInfosOwned_slimmedJets_*_*`, replacing the `tagInfoSources.append( cms.InputTag("pixelClusterTagInfos") )` with `tagInfoSources = cms.VInputTag( cms.InputTag("pixelClusterTagInfos") )` in `miniAOD_tools.py`. We also implemented the new accessor method in Jet.h and Jet.cc.